### DISCUSSION TITLE
PuerTS可以在UWP平台上工作吗？

### DISCUSSION THREAD

**Discussion by QinmingRen** [Tencent/puerts#1255] [Q&A]
公司产品希望在Hololens2上使用Unity + PuerTS方案。Hololens2是Arm64 + UWP环境，目前PuerTS并不能支持。QuickJS看起来可以作为保底方案，不太清楚它和V8的兼容性怎样，请问大家有没有人尝试过在Arm64 + UWP环境下编译V8呢？

**Comment by chexiongsheng** [comment]
可以试试这个项目：https://github.com/bnoordhuis/v8-cmake ，看维护人员有nodejs的人。
这个是v8的cmake编译支持。cmake支持uwp。
要注意这个仓库并不是每个版本都是好的，我试过一些版本在linux编译就失败。亲测这个版本至少linux是可用的：
https://github.com/bnoordhuis/v8-cmake/commit/8dfb23f20c466f290d454c4cd63ef93baa49765c

**Reply by QinmingRen** [reply]
谢谢，我试试看

**Reply by QinmingRen** [reply]
对了，Hololens2的环境比较奇葩，是Windows + Arm64 + UWP

**Comment by chexiongsheng** [comment]
QuickJS版本在Unity下兼容性还好，有项目在用，没啥功能方面的问题。
主要是性能，性能一方面虚拟机可能不如关闭jit后的v8（不知道Hololens2能让jit不），其次经过多层封装，跨语言调用也可能比v8版本差（不过后面puerts v2支持quickjs后可以优化，优化后有可能比v8跨语言还好些）
